src/controllers/commentController.js

```diff
@@ -0,0 +1,96 @@
+import commentService from '../services/commentService.js';
+import mongoose from 'mongoose';
+
+export const createComment = async (req, res) => {
+  try {
+    console.log('Server:: create the Comment');
+    const id = req.params.threadId;
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+      return res.status(400).send({
+        message: 'Invalid thread id',
+        data: {}
+      });
+    }
+    const comment = await commentService.createComment(id, req.params, req.body);
+    res.status(201).json({
+      message: 'The comment is created successfully',
+      commentCreated: comment
+    });
+  } catch (err) {
+    console.error(err);
+    res.status(400).send('Cannot create this comment');
+  }
+};
+
+export const getComment = async (req, res) => {
+  try {
+    console.log('Server::get the comment');
+    const id = req.params.id;
+    const comment = await commentService.getComment(id);
+    res.status(200).json({ comment });
+  } catch (err) {
+    res.status(400).json({
+      error: err.message
+    });
+  }
+};
+
+export const getCommentsByThread = async (req, res) => {
+  try {
+    console.log('Server::Getting comments - running getComments');
+    const threadId = req.params.threadId;
+    if (!mongoose.Types.ObjectId.isValid(threadId)) {
```

Comment:
again.

Reply:
If removed, the postman has an error like "error": "Cast to ObjectId failed for value \"6483f0e32215626486c35\" (type string) at path \"_id\" for model \"Thread\". I think it is more clear in this way.
}